src/core/compact_object.cc

```diff
@@ -696,6 +697,8 @@ void CompactObj::SetJson(JsonType&& j) {
     SetMeta(JSON_TAG);
     u_.json_obj.json_len = j.size();
     u_.json_obj.json_ptr = AllocateMR<JsonType>(std::move(j));
+    DCHECK(std::pmr::polymorphic_allocator<char>{memory_resource()} ==
```

Comment:
Bingo. 

```
2024-09-03T16:48:59.0738486Z 60: F20240903 16:48:59.073283 22070 compact_object.cc:700] Check failed: std::pmr::polymorphic_allocator<char>{memory_resource()} == u_.json_obj.json_ptr->get_allocator() 
2024-09-03T16:48:59.0740855Z 60: *** Check failure stack trace: ***
2024-09-03T16:48:59.0761492Z 60:     @     0x564941768dbb  google::LogMessage::Fail()
2024-09-03T16:48:59.0781948Z 60:     @     0x564941768d02  google::LogMessage::SendToLog()
2024-09-03T16:48:59.0802555Z 60:     @     0x5649417684f7  google::LogMessage::Flush()
2024-09-03T16:48:59.0824658Z 60:     @     0x56494176c20e  google::LogMessageFatal::~LogMessageFatal()
2024-09-03T16:48:59.0869710Z 60:     @     0x5649412c6855  dfly::CompactObj::SetJson()
2024-09-03T16:48:59.0895859Z 60:     @     0x564940be862b  dfly::RdbLoaderBase::OpaqueObjLoader::HandleBlob()
2024-09-03T16:48:59.0930073Z 60:     @     0x564940be4b97  dfly::RdbLoaderBase::OpaqueObjLoader::operator()()
2024-09-03T16:48:59.0972643Z 60:     @     0x564940c0b783  std::__invoke_impl<>()
2024-09-03T16:48:59.1012552Z 60:     @     0x564940c05aaa  std::__invoke<>()
2024-09-03T16:48:59.1059136Z 60:     @     0x564940bffb53  std::__detail::__variant::__gen_vtable_impl<>::__visit_invoke_impl()
2024-09-03T16:48:59.1091480Z 60:     @     0x564940bffb96  std::__detail::__variant::__gen_vtable_impl<>::__do_visit_invoke()
```

`memory_resource was not propagated`.

Reply:
From my understanding the allocate used should have been `CompactObject::memory_resource()`. I haven't checked the specific case but based on the semantics on the json family this should be the case. Let me know if I am wrong.

Reply:
If you think this is a problem with jsoncons, I'd try to create a simple standalone example of this happening and report it in their repo.
We've already reported similar issues before, and Daniel (the maintainer) always helped us